dev-packages/cli/src/download-plugins.spec.ts

```diff
@@ -0,0 +1,222 @@
+/********************************************************************************
+ *
+ * This program and the accompanying materials are made available under the
```

Comment:
The following license header is incorrect, it should follow the format like:

https://github.com/eclipse-theia/theia/blob/ae35f9c1a4bf0127f8b15262422da93ad5a13ac0/packages/core/src/browser/about-dialog.tsx#L1-L15

If you do not work for a company, you should use your name instead.

Reply:
Thanks for feedback !

> @perrinjerome the following proposal would require discussions by the community on what the behavior should be, and if we should introduce such a `lockfile` for plugins.

Yes, definitely. These changes are an "example" that could help the discussion.

> Currently, there are a couple of issues with this approach:
> 
> * We should not impose the `lockfile`, it should be possible to not include/generate it through a flag (since this is the current behavior, it is possible that the community will want this as default).

This make sense.

> * The current approach does not work since it is possible to have `.vsix` and other types of packaged plugins which are not under `openvsx` but can be from github releases (like our [theia-apps](https://github.com/theia-ide/theia-apps/blob/645ce4617d105296554f11adb3633180c73bc357/theia-docker/latest.package.json#L105-L106) example. It is also possible to mix and match where we fetch plugins from which leads to bogus `lockfiles` (since it is assumed that the location remains the same).

This implementation keeps support for using URLs in package.json like it is done today, so in package json we can use either URLs or a special `publisher/extension@version`.

> It is also bogus to assume that the only registry is `openvsx` for which we fetch plugins.

There was a command line flag to specify another registry base URL, for cases of self hosted openvsx. But this assumes that the registry has opensvx API ( https://open-vsx.org/swagger-ui/#/registry-api/getExtensionUsingGET )